pkg/etcd/etcd.go

```diff
@@ -904,9 +904,7 @@ func (e *ETCD) listSnapshots(ctx context.Context, snapshotDir string) ([]snapsho
 		ctx, cancel := context.WithCancel(ctx)
 		defer cancel()
 
-		objects := e.s3.client.ListObjects(ctx, e.config.EtcdS3BucketName, minio.ListObjectsOptions{
-			Prefix: e.s3.snapshotPrefix(),
-		})
+		objects := e.s3.client.ListObjects(ctx, e.config.EtcdS3BucketName, minio.ListObjectsOptions{})
```

Comment:
How does this work on a bucket with many objects? @ibuildthecloud mentioned that folks frequently dump backups into a bucket with a bunch of other stuff; there could potentially be thousands of objects here.

Reply:
Given the time frame I haven't had the opportunity to test at that scale. In my own bucket, I've tested with a few hundred and there's nothing different than just a couple.